From 87a64ea3d7259f3ac32a6f70a6c6bf0ab192fc33 Mon Sep 17 00:00:00 2001
From: Jilles Tjoelker <jilles@stack.nl>
Date: Thu, 27 May 2010 20:07:29 +1000
Subject: [EXPAND] Fix corruption of redirections with byte 0x81

In other ash variants, a partial implementation of ksh-like cmd >file*
adds and removes CTLESC bytes ('\x81') in redirection filenames,
preserving 8-bit transparency. Long ago, dash removed the code to add
the CTLESC bytes, but not the code to remove them, causing corruption of
filenames containing CTLESC. This commit removes the code to remove the
CTLESC bytes.

The CTLESC byte occurs frequently in UTF-8 encoded non-Latin text.

This bug has been reported various times to Ubuntu and Debian (e.g.
Launchpad Ubuntu #422298). This patch is the same as the one submitted
by Alexander Korolkov in Ubuntu #422298.

Signed-off-by: Jilles Tjoelker <jilles@stack.nl>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit f8231aea37e921492fc7fbd972385ab5b90e8627)

Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 ChangeLog |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 2bf7d95..5b0bb5a 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,6 +1,7 @@
 2010-05-27  Jilles Tjoelker <jilles@stack.nl>
 
 	* Force fork if any trap is set, not just on EXIT.
+	* Fix corruption of redirections with byte 0x81.
 
 2010-05-27  Herbert Xu <herbert@gondor.apana.org.au>
 
-- 
1.7.4.1

